subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -207,7 +216,12 @@ private OperationGroup(String category, @Nullable String loggingHeader, String d
         }
 
         private StyledTextOutputEvent header() {
-            return new StyledTextOutputEvent(lastUpdateTime, category, null, buildOpIdentifier, headerFormatter.format(loggingHeader, description, shortDescription, status));
+            List<StyledTextOutputEvent.Span> spans = Lists.newArrayList();
+            if (didHaveOutput) {
+                spans.add(EOL);
+            }
+            spans.addAll(headerFormatter.format(loggingHeader, description, shortDescription, status));
```

Comment:
I wonder if we need to be careful with the amount of garbage we may be creating now.

We create a 2 new lists here on every header and only sometimes need the second one.